src/util/ConfigManager/ConfigOption.h

```diff
@@ -228,52 +214,18 @@ class ConfigOption {
   example: "The amount of rows in the table. Has a default value of 3."
   @param variablePointer The variable, this pointer points to, will always be
   overwritten by any of the set functions of this class.
-  @param defaultValue The default value.
+  @param defaultValue The optional default value. An empty `std::optional` is
+  for no default value and a non empty for a default value.
   */
   template <typename T>
   requires ad_utility::isTypeContainedIn<T, ConfigOption::AvailableTypes>
-  ConfigOption(std::string_view identifier, std::string_view description,
-               T* variablePointer, const T& defaultValue)
-      : data_{Data<T>{std::optional<T>(defaultValue), variablePointer}},
+  ConfigOption(
+      std::string_view identifier, std::string_view description,
+      T* variablePointer,
+      const std::optional<T>& defaultValue = std::optional<T>(std::nullopt))
+      : data_{Data<T>{defaultValue, variablePointer}},
```

Comment:
I'm very much against such behavior.

First of all, the move is not clear from outside. Which could lead to all sort of undefined behavior.

Secondly, we can no longer use r-values for default values. A lot of the time, we simply want to use a r-value and not have to defined an extra variable, just so that we can pass that r-value.

Reply:
Turns out, there was a typo in the suggestion and the actual suggestion makes much more sense. Implemented in [2917388](https://github.com/ad-freiburg/qlever/pull/977/commits/291738836b4b0035b997e79d546ab376f7997b75).